internal/sdkprovider/service/kafkatopic/kafka_topic.go

```diff
@@ -269,18 +269,11 @@ func resourceKafkaTopicCreate(ctx context.Context, d *schema.ResourceData, m int
 	// aiven.KafkaTopics.Create() function may return 501 on create
 	// Second call might say that topic already exists
 	// So to be sure, better check it before create
-	_, err := getTopic(ctx, m, d.Timeout(schema.TimeoutRead), project, serviceName, topicName)
-
-	// No error means topic exists
-	if err == nil {
+	client := m.(*aiven.Client)
+	if isTopicExists(ctx, client, project, serviceName, topicName) {
```

Comment:
consider adding a comment above this block to briefly explain its purpose for clarity